"gobgp neighbor A.B.C.D local" command cause crash of gobgpd

"gobgp neighbor A.B.C.D local"

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x4a53ac]

goroutine 186 [running]:
panic(0xa889c0, 0xc42000c040)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/osrg/gobgp/api.(*Server).GetRib(0xc42012c980, 0x7f7d357a1000, 0xc4203ae3c0, 0xc42008a018, 0x0, 0x18, 0x18)
        /home/kunihiro/src/github.com/osrg/gobgp/api/grpc_server.go:315 +0x16c
github.com/osrg/gobgp/api._GobgpApi_GetRib_Handler(0xb53460, 0xc42012c980, 0x7f7d357a1000, 0xc4203ae3c0, 0xc42006a240, 0x0, 0x0, 0x0, 0x300000002, 0xc420026c00)
        /home/kunihiro/src/github.com/osrg/gobgp/api/gobgp.pb.go:4936 +0x280
google.golang.org/grpc.(*Server).processUnaryRPC(0xc4201fc0c0, 0x1088e80, 0xc4201c8180, 0xc42014e200, 0xc420178e40, 0x10751e8, 0x0, 0x0, 0x0)
        /home/kunihiro/src/google.golang.org/grpc/server.go:708 +0xbd8
google.golang.org/grpc.(*Server).handleStream(0xc4201fc0c0, 0x1088e80, 0xc4201c8180, 0xc42014e200, 0x0)
        /home/kunihiro/src/google.golang.org/grpc/server.go:892 +0x6a4
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc4202642c0, 0xc4201fc0c0, 0x1088e80, 0xc4201c8180, 0xc42014e200)
        /home/kunihiro/src/google.golang.org/grpc/server.go:468 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/kunihiro/src/google.golang.org/grpc/server.go:469 +0xa3

--- a/server/server.go
+++ b/server/server.go
@@ -993,7 +993,7 @@ func (s *BgpServer) AddBmp(c *config.BmpServerConfig) error {
 	}, true)
 }
 
-func (s *BgpServer) DeleteBmp(c *config.BmpServerConfig) (err error) {
+func (s *BgpServer) DeleteBmp(c *config.BmpServerConfig) error {
 	return s.mgmtOperation(func() error {
 		return s.bmpManager.deleteServer(c)
 	}, true)
@@ -1133,7 +1133,7 @@ func (server *BgpServer) fixupApiPath(vrfId string, pathList []*table.Path) erro
 }
 
 func (s *BgpServer) AddPath(vrfId string, pathList []*table.Path) (uuidBytes []byte, err error) {
-	s.mgmtOperation(func() error {
+	err = s.mgmtOperation(func() error {
 		if err := s.fixupApiPath(vrfId, pathList); err != nil {
 			return err
 		}
@@ -1401,7 +1401,7 @@ func (s *BgpServer) SoftReset(addr string, family bgp.RouteFamily) error {
 }
 
 func (s *BgpServer) GetRib(addr string, family bgp.RouteFamily, prefixes []*table.LookupPrefix) (rib *table.Table, err error) {
-	s.mgmtOperation(func() error {
+	err = s.mgmtOperation(func() error {
 		m := s.globalRib
 		id := table.GLOBAL_RIB_NAME
 		if len(addr) > 0 {
@@ -1426,7 +1426,7 @@ func (s *BgpServer) GetRib(addr string, family bgp.RouteFamily, prefixes []*tabl
 }
 
 func (s *BgpServer) GetVrfRib(name string, family bgp.RouteFamily, prefixes []*table.LookupPrefix) (rib *table.Table, err error) {
-	s.mgmtOperation(func() error {
+	err = s.mgmtOperation(func() error {
 		m := s.globalRib
 		vrfs := m.Vrfs
 		if _, ok := vrfs[name]; !ok {
@@ -1452,7 +1452,7 @@ func (s *BgpServer) GetVrfRib(name string, family bgp.RouteFamily, prefixes []*t
 }
 
 func (s *BgpServer) GetAdjRib(addr string, family bgp.RouteFamily, in bool, prefixes []*table.LookupPrefix) (rib *table.Table, err error) {
-	s.mgmtOperation(func() error {
+	err = s.mgmtOperation(func() error {
 		peer, ok := s.neighborMap[addr]
 		if !ok {
 			return fmt.Errorf("Neighbor that has %v doesn't exist.", addr)
@@ -1474,7 +1474,7 @@ func (s *BgpServer) GetAdjRib(addr string, family bgp.RouteFamily, in bool, pref
 }
 
 func (s *BgpServer) GetRibInfo(addr string, family bgp.RouteFamily) (info *table.TableInfo, err error) {
-	s.mgmtOperation(func() error {
+	err = s.mgmtOperation(func() error {
 		m := s.globalRib
 		id := table.GLOBAL_RIB_NAME
 		if len(addr) > 0 {
@@ -1494,7 +1494,7 @@ func (s *BgpServer) GetRibInfo(addr string, family bgp.RouteFamily) (info *table
 }
 
 func (s *BgpServer) GetAdjRibInfo(addr string, family bgp.RouteFamily, in bool) (info *table.TableInfo, err error) {
-	s.mgmtOperation(func() error {
+	err = s.mgmtOperation(func() error {
 		peer, ok := s.neighborMap[addr]
 		if !ok {
 			return fmt.Errorf("Neighbor that has %v doesn't exist.", addr)
@@ -1680,7 +1680,7 @@ func (s *BgpServer) DeleteNeighbor(c *config.Neighbor) error {
 }
 
 func (s *BgpServer) UpdateNeighbor(c *config.Neighbor) (policyUpdated bool, err error) {
-	s.mgmtOperation(func() error {
+	err = s.mgmtOperation(func() error {
 		addr := c.Config.NeighborAddress
 		peer, ok := s.neighborMap[addr]
 		if !ok {
@@ -1842,7 +1842,7 @@ func (s *BgpServer) DisableNeighbor(addr string) error {
 }
 
 func (s *BgpServer) GetDefinedSet(typ table.DefinedType) (sets *config.DefinedSets, err error) {
-	s.mgmtOperation(func() error {
+	err = s.mgmtOperation(func() error {
 		sets, err = s.policy.GetDefinedSet(typ)
 		return nil
 	}, false)
@@ -1945,7 +1945,7 @@ func (server *BgpServer) toPolicyInfo(name string, dir table.PolicyDirection) (s
 }
 
 func (s *BgpServer) GetPolicyAssignment(name string, dir table.PolicyDirection) (rt table.RouteType, l []*config.PolicyDefinition, err error) {
-	s.mgmtOperation(func() error {
+	err = s.mgmtOperation(func() error {
 		var id string
 		id, err = s.toPolicyInfo(name, dir)
 		if err != nil {
@@ -2021,7 +2021,7 @@ func (s *BgpServer) ValidateRib(prefix string) error {
 }
 
 func (s *BgpServer) GetRpki() (l []*config.RpkiServer, err error) {
-	s.mgmtOperation(func() error {
+	err = s.mgmtOperation(func() error {
 		l = s.roaManager.GetServers()
 		return nil
 	}, false)

